File: android/src/main/java/com/osmdroid/TileCacherModule.java
package com.osmdroid;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.osmdroid.OsmMapTileCacher;

public class TileCacherModule extends ReactContextBaseJavaModule {

    private final OsmMapTileCacher cacher;

    public TileCacherModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.cacher = new OsmMapTileCacher(reactContext);
    }

    @Override
    public String getName() {
        return "TileCacher";
    }

    @ReactMethod
    public void cacheTilesFromDirectory(String directoryPath) {
        cacher.cacheTilesFromDirectory(directoryPath);
    }
}

File: android/src/main/java/com/osmdroid/TileCacherModule.kt
package com.osmdroid;

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap

class TileCacherModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "TileCacher"
    }

    @ReactMethod
    fun cacheTilesFromDirectory(directoryPath: String, options: ReadableMap?) {
        val showProgressToast = options?.getBoolean("showProgressToast") ?: false
        OsmMapTileCacher(reactApplicationContext).cacheTilesFromDirectory(directoryPath, showProgressToast)
    }
}